src/dev/flang/ast/InlineArray.java

```diff
@@ -124,14 +124,7 @@ AbstractType typeForInferencing()
   {
     if (_type == null && !_elements.isEmpty())
       {
-        AbstractType t = Types.resolved.t_void;
-        for (var e : _elements)
-          {
-            var et = e.typeForInferencing();
-            t =
-              t  == null ? null :
-              et == null ? null : t.union(et);
-          }
+        var t = Expr.union(_elements, false);
```

Comment:
Can you add a comment explaining when `ignoreUnknown` does not work here? I could imagine an array as follows
```
  f i32->i32 := x->x+1
  a := [f,f,x->x+2]
```
Where type inference for `a` could work fine if `itnoreUnknown`  was `true`.

Reply:
I changed ignoreUnkown to `true` but example does not work yet due to other limitations.